### PR TITLE
creating new service named cronus, proxying hyperscalar statless serv…

### PR DESCRIPTION
**Pull request by dhalimi**
…ices, introducing the AWS SES - simple email service

### CODE REVIEW

File: openstack/cronus/templates/cronus/_config.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    http: # default :5000
+    smtp: # default :1025
+  keystone:
+    authUrl: {{ .Values.config.authUrl }}
+    applicationCredentialID: {{ .Values.config.applicationCredentialID }}
+    applicationCredentialSecret: {{ .Values.config.applicationCredentialSecret }}
+    region: {{ .Values.config.region }}
+    endpointType: {{ .Values.config.endpointType }}
+    verifyEmailDomain: example.com
+    verifyEmailSecret: domainkey
```
Comment (kayrus) [inline openstack/cronus/templates/cronus/_config.yaml.tpl:19]: Can you parametrize above two values? Besides these values have to be placed under the `aws` in `openstack/cronus/templates/nebula/_config.yaml.tpl`.

File: openstack/cronus/templates/cronus/_config.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.cronus.enabled -}}
+cronus:
+  cacheSize: {{ .Values.cronus.cacheSize }}
+  billingCacheTTL: {{ .Values.config.billingCacheTTL }}
+  aws:
+    forwardUserAgent: {{ .Values.config.forwardUserAgent }}
+    allowedServices:
+      email: {{ .Values.config.allowedServices.email }}
```
Comment (kayrus) [inline openstack/cronus/templates/cronus/_config.yaml.tpl:8]: What if we need to add another stateless service? I suggest:
```
allowedServices: {{ .Values.config.allowedServices }}
```

File: openstack/cronus/templates/cronus/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{{- if .Values.cronus.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
```
Comment (kayrus) [inline openstack/cronus/templates/cronus/deployment.yaml:3]: Didn't we agree that we'll use statefulset?

File: openstack/cronus/templates/cronus/_config.yaml.tpl [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  listenAddr:
+    http: # default :5000
+    smtp: # default :1025
+  keystone:
```
Comment (kayrus) [inline openstack/cronus/templates/cronus/_config.yaml.tpl:12]: We'll probably use a technical user in future, therefore I suggest to parametrize all possible auth parameters, not only application credentials. Besides, if application credentials are not defined, this shouldn't cause an error.

File: openstack/cronus/templates/nebula/_config.yaml.tpl [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{{- if .Values.nebula.enabled -}}
+nebula:
+  cacheSize: {{ .Values.nebula.cacheSize }}
+  listenAddr:
+    http: # default :1080
+  keystone:
```
Comment (kayrus) [inline openstack/cronus/templates/nebula/_config.yaml.tpl:6]: We'll probably use a technical user in future, therefore I suggest to parametrize all possible auth parameters, not only application credentials. Besides, if application credentials are not defined, this shouldn't cause an error.

File: openstack/cronus/templates/nebula/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{{- if .Values.nebula.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
```
Comment (kayrus) [inline openstack/cronus/templates/nebula/deployment.yaml:3]: Didn't we agree that we'll use statefulset?

File: openstack/cronus/templates/nebula/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      http:
+        paths:
+          - backend:
+              serviceName: nebula-service
+              servicePort: {{ .Values.nebula.http }}
+            path: /
+  tls:
+    - hosts:
+      - nebula.{{ .Values.config.region }}.cloud.sap
+      secretName: tls-secret-nebula-{{ .Values.config.region }}-cloud-sap
```
Comment (kayrus) [inline openstack/cronus/templates/nebula/ingress.yaml:24]: can we unify a secret name? e.g. use only `tls-secret-nebula` as a name

File: openstack/cronus/values.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  applicationCredentialID: application-credential-id
+  applicationCredentialSecret: application-credential-secret
+  awsAccess: aws-access
+  awsSecret: aws-secret
+  awsUsername: aws-username
+  awsPolicy: aws-policy
+  endpointType: public
+  forwardUserAgent: false
+  billingCacheTTL: 24h
+  technicalResponsible: technical-responsible
```
Comment (kayrus) [inline openstack/cronus/values.yaml:45]:
```suggestion
  technicalResponsible: technical-responsible@example.com
```

File: openstack/cronus/templates/cronus/ingress.yaml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+      http:
+        paths:
+          - backend:
+              serviceName: cronus-service
+              servicePort: {{ .Values.cronus.http }}
+            path: /
+  tls:
+    - hosts:
+      - cronus.{{ .Values.config.region }}.cloud.sap
+      secretName: tls-secret-cronus-{{ .Values.config.region }}-cloud-sap
```
Comment (kayrus) [inline openstack/cronus/templates/cronus/ingress.yaml:24]: can we unify a secret name? e.g. use only `tls-secret-cronus` as a name

**Review by kayrus — CHANGES_REQUESTED** [review]

File: openstack/cronus/values.yaml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  group: service-group
+  jiraUsername: jira-username
+  jiraPassword: jira-password
+  serviceUsername: service-username
+  servicePassword: service-password
+  applicationCredentialID: application-credential-id
+  applicationCredentialSecret: application-credential-secret
+  awsAccess: aws-access
+  awsSecret: aws-secret
+  awsUsername: aws-username
```
Comment (kayrus) [inline openstack/cronus/values.yaml:40]:
```suggestion
  technicalUsername: aws-username
```

File: openstack/cronus/templates/cronus/_config.yaml.tpl (hunk 2 above)
Comment (dhalimi) [reply openstack/cronus/templates/cronus/_config.yaml.tpl:8]: I added a range 
allowedServices:
    {{- range $key, $value := .Values.config.allowedServices }}
      {{ $key }}: {{ $value }}
    {{- end }}

File: openstack/cronus/templates/cronus/deployment.yaml (hunk 3 above)
Comment (dhalimi) [reply openstack/cronus/templates/cronus/deployment.yaml:3]: {{- if .Values.cronus.enabled }}
apiVersion: apps/v1
{{ if .Values.cronus.statefulSet.enabled }}
kind: StatefulSet
{{ else }}
kind: Deployment
{{ end -}}

File: openstack/cronus/templates/cronus/_config.yaml.tpl (hunk 4 above)
Comment (dhalimi) [reply openstack/cronus/templates/cronus/_config.yaml.tpl:12]:
```
  listenAddr:
    http: :{{ .Values.cronus.port.http }} # default :5000
    smtp: :{{ .Values.cronus.port.smtp }} # default :1025
```

File: openstack/cronus/templates/nebula/_config.yaml.tpl (hunk 5 above)
Comment (dhalimi) [reply openstack/cronus/templates/nebula/_config.yaml.tpl:6]:
```
  listenAddr:
    http: :{{ .Values.nebula.port.http }} # default :1080
```

File: openstack/cronus/templates/nebula/deployment.yaml (hunk 6 above)
Comment (dhalimi) [reply openstack/cronus/templates/nebula/deployment.yaml:3]:
```
{{- if .Values.nebula.enabled }}
apiVersion: apps/v1
{{ if .Values.nebula.statefulSet.enabled }}
kind: StatefulSet
{{ else }}
kind: Deployment
{{ end -}}
metadata:
```

File: openstack/cronus/templates/nebula/ingress.yaml (hunk 7 above)
Comment (dhalimi) [reply openstack/cronus/templates/nebula/ingress.yaml:24]:
```
  tls:
    - hosts:
      - nebula.{{ .Values.config.region }}.cloud.sap
      secretName: tls-secret-nebula
```

File: openstack/cronus/templates/cronus/ingress.yaml (hunk 9 above)
Comment (dhalimi) [reply openstack/cronus/templates/cronus/ingress.yaml:24]:
```
    - hosts:
      - cronus.{{ .Values.config.region }}.cloud.sap
      secretName: tls-secret-cronus
```

**Pull request merged**